packages/unenv-preset/tests/index.test.ts

```diff
@@ -0,0 +1,62 @@
+import fs from "node:fs";
+import path from "node:path";
+import { platform } from "node:process";
+import { fileURLToPath } from "node:url";
+import { fetch } from "undici";
+import { afterAll, beforeAll, describe, expect, test } from "vitest";
+import { runWranglerDev } from "../../../fixtures/shared/src/run-wrangler-long-lived";
+import { TESTS } from "./worker/index";
+
+const pkgDir = path.resolve(fileURLToPath(import.meta.url), "../..");
+// workerd binary
+const workerdBin = platform === "win32" ? "workerd.ps1" : "workerd";
+const localWorkerdPath = path.join(pkgDir, "node_modules/.bin/", workerdBin);
+// wrangler binary
+const wranglerPath = path.join(pkgDir, "../wrangler/bin/wrangler.js");
+// Base path for resolving `@cloudflare/unenv-preset` files
+const localPresetResolveBaseDir = path.join(pkgDir, "package.json");
+// Base path for resolving `unjs/unenv` files
+const localUnenvResolveBaseDir = path.join(
+	pkgDir,
+	"node_modules/unenv/package.json"
+);
+
+describe(`@cloudflare/unenv-preset ${platform} ${localWorkerdPath} ${fs.existsSync(localWorkerdPath)}`, () => {
+	let wrangler: Awaited<ReturnType<typeof runWranglerDev>>;
+
+	beforeAll(async () => {
+		// Use workerd binary install in `@cloudflare/unenv-preset`
+		// rather than the one bundled with wrangler.
+		const MINIFLARE_WORKERD_PATH = localWorkerdPath;
+		// Use the preset from the local `@cloudflare/unenv-preset` and `unjs/unenv`
+		// rather than the one bundled with wrangler.
+		const WRANGLER_UNENV_RESOLVE_PATHS = [
+			localPresetResolveBaseDir,
+			localUnenvResolveBaseDir,
+		].join(",");
+
+		console.log({ MINIFLARE_WORKERD_PATH, WRANGLER_UNENV_RESOLVE_PATHS });
+
+		console.log(path.resolve(__dirname, "../"));
+
+		wrangler = await runWranglerDev(
+			path.resolve(__dirname, "../"),
+			["--port=0", "--inspector-port=0"],
+			{
+				MINIFLARE_WORKERD_PATH,
+				WRANGLER_UNENV_RESOLVE_PATHS,
+			}
+		);
+	}, 30000);
+
+	afterAll(async () => {
+		await wrangler.stop();
```

Comment:
It's possible that this could be undefined if `beforeAll()` fails? So perhaps

```suggestion
		await wrangler?.stop();
```

which should solve the types error?

Reply:
I had `let wrangler: ... | undefined;` first but then `const { ip, port } = wrangler;` complains because it might be undefined.
Because tests will fail anyway after a timeout I left as it but let me know if you think I should change.

Reply:
I made the update